generator/src/main/java/com/graphicsfuzz/generator/fuzzer/OpaqueExpressionGenerator.java

```diff
@@ -672,10 +672,11 @@ public Expr apply(Expr expr, BasicType type, boolean constContext, int depth,
       // (false ? whatever : expr)
       // or
       // (true  ? expr : whatever)
-      assert BasicType.allScalarTypes().contains(type);
-      // Only generate ternary expressions for scalars; the vector case causes massive blow-up
+      assert BasicType.allNumericTypes().contains(type);
       Expr exprWithIdentityApplied = applyIdentityFunction(expr, type, constContext, depth, fuzzer);
-      Expr something = fuzzedConstructor(fuzzer.fuzzExpr(type, false, constContext, depth));
+      // We have to make sure that the LHS and RHS of the ternary expression evaluate to the same
+      // type, so we set constContext to true to ensure no side effects.
+      Expr something = fuzzedConstructor(fuzzer.fuzzExpr(type, false, true, depth));
```

Comment:
I don't understand why you have changed this from what the code did before.  The point of "something" is that it doesn't actually get executed (it is in the unreachable branch of the ternary), so it is OK for it to have side-effects.  Thus constContext should match whatever it was coming in, as in the original code.

If this leads to test failures then there must be something else going on, which we should investigate.  Whether constContext holds or not should not affect the type of the generated expression.

Reply:
Sorry, I forgot to revert that when we decided to fork the constructor identity into a different PR. The reason tests were failing before is because the fuzzer occasionally seemed to produce expressions that would not resolve to the same type as the vector (producing only scalar expressions instead of matrix or vector expressions).